mlflow/transformers/model_io.py

```diff
@@ -0,0 +1,182 @@
+
+    Args:
+        path: The local path to save the pipeline
+        pipeline: Transformers pipeline instance
+        flavor_config: The flavor configuration constructed for the pipeline
+        processor: Optional processor instance to save alongside the pipeline
+    """
+    pipeline.model.save_pretrained(
+        save_directory=path.joinpath(_MODEL_BINARY_FILE_NAME),
+        max_shard_size=MLFLOW_HUGGINGFACE_MODEL_MAX_SHARD_SIZE.get(),
+    )
+
+    component_dir = path.joinpath(_COMPONENTS_BINARY_DIR_NAME)
+    for name in flavor_conf.get(FlavorKey.COMPONENTS, []):
+        getattr(pipeline, name).save_pretrained(component_dir.joinpath(name))
+
+    if processor:
+        processor.save_pretrained(component_dir.joinpath(_PROCESSOR_BINARY_DIR_NAME))
+
+
+def load_model_and_components_from_local(path, flavor_conf, accelerate_conf, device=None):
+    """
+    Load the model and components of a Transformer pipeline from the specified local path.
+
+    Args:
+        path: The local path contains MLflow model artifacts
+        flavor_conf: The flavor configuration
+        accelerate_conf: The configuration for the accelerate library
+        device: The device to load the model onto
+    """
+    loaded = {}
+
+    # NB: Path resolution for models that were saved prior to 2.4.1 release when the patching for
+    #     the saved pipeline or component artifacts was handled by duplicate entries for components
+    #     (artifacts/pipeline/* and artifacts/components/*) and pipelines were saved via the
+    #     "artifacts/pipeline/*" path. In order to load the older formats after the change, the
+    #     presence of the new path key is checked.
+    model_path = path.joinpath(flavor_conf.get(FlavorKey.MODEL_BINARY, "pipeline"))
+    loaded[FlavorKey.MODEL] = _load_model(model_path, flavor_conf, accelerate_conf, device)
+
+    components = flavor_conf.get(FlavorKey.COMPONENTS, [])
+    if FlavorKey.PROCESSOR_TYPE in flavor_conf:
+        components.append("processor")
+
+    for component_key in components:
+        loaded[component_key] = _load_component(flavor_conf, component_key, local_path=path)
+
+    return loaded
+
+
+def load_model_and_components_from_huggingface_hub(flavor_conf, accelerate_conf, device=None):
+    """
+    Load the model and components of a Transformer pipeline from HuggingFace Hub.
+
+    Args:
+        flavor_conf: The flavor configuration
+        accelerate_conf: The configuration for the accelerate library
+        device: The device to load the model onto
+    """
+    loaded = {}
+
+    model_repo = flavor_conf[FlavorKey.MODEL_NAME]
+    model_revision = flavor_conf.get(FlavorKey.MODEL_REVISION, None)
+
+    if not model_revision:
+        _logger.warn(
+            "It seems the specified model is saved with 'save_pretrained' set to False, "
+            "but the model commit hash is not found in the saved configuration. MLflow will "
```

Comment:
When/how does this happen?

Reply:
replaced with an exception, since this is pretty much unexpected thing to happen